### ISSUE TITLE
Add the manager / youtube scripts only at component level

### ISSUE THREAD

**Issue by suterma** [vue-youtube/vue-youtube#8]
### Feature request

This component, together with the manager plugin works great, but I have usability/performance/privacy issues: 

The manager loads quite a bunch from youtube and in case the Vue application does not play or offer to play any videos at startup, these are unnecessary for the moment.

Thus, I would like to have the option to not register the manager plugin prematurely, at the app level, but only later, upon user consent. Although it's quite obvious, I would like to inform the users of my app about this data disclosure to youtube when they paste a Youtube URL into my app.

### A possible solution

I have successfully registered the manager at the component level. It requires an internal method, but seems to work stable so far:

```
<script setup lang="ts">
// Import the 'usePlayer' function
import { usePlayer } from '@vue-youtube/core';
import { onMounted, ref } from 'vue';
import { getCurrentInstance } from 'vue';
import { createManager } from '@vue-youtube/core';

// Locally register the create manager plugin (avoid vue instance pollution and
// data disclosure, when this component is not used)
const app = getCurrentInstance();
if (app) {
    app.appContext.app.use(createManager());
}

// Use a template ref to reference the target element
const player = ref();

// Call the 'usePlayer' function with the desired video ID and target ref
const { onReady } = usePlayer('sKGoqpqJ-MM', player, { //....
```
**However, a better solution would be an option or other possibility in the manager itself, to defer loading of said scripts.**

### A better solution

- provide an "enabled later" option to the manager (off by default), then in the manager do not yet call the `_insert` method. 
- Later, each component that actually wants to load a Youtube video, calls the manager's new `enable()` method, that actually enables the player (guarded, to actually execute only once on the document/window).

Since you seem to inject the manager to each player component already, this should be not a great deal.

**Comment by Techassi** [comment]
I currently don't have to read through this in detail, but I will come back to you in a few days. But from a quick look it sounds like a nice idea.

**Comment by suterma** [comment]
Looks great!. I will test this in my app asap.

**Comment by suterma** [comment]
Sorry to say, but I fail to compile my app with the new version. I get weird compilation errors (I use typescript), but mostly, the new load() function seems not to be defined in the `manager.d.ts` file.

I see 

```
export interface Manager {
    install(app: App): void;
    register(target: HTMLElement, cb: RegisterFunction): void;
    state: ManagerState;
    _insert(): void;
}
```
in the module's dist folder. Maybe your build process did not work to the end?

**Comment by Techassi** [comment]
Okay this seems weird. I will check this asap!

**Comment by Techassi** [comment]
Did you update to version `0.0.5`? The new release doesn't include a `manager.d.ts` file anymore. Instead there only should be one `index.d.ts` file with all types merged into it.

**Comment by suterma** [comment]
I did update. And I took this code from https://www.npmjs.com/package/@vue-youtube/core?activeTab=code then the manager.d.ts in /@vue-youtube/core/dist/core/manager.d.ts 

I did not find the load method in dist/index.js nor in core/indext.d.ts

**Comment by Techassi** [comment]
Okay, I looked into this a little more. There is something weird happening...

My Local build produces these files:

```
ls dist
index.cjs
index.d.ts
index.js
```

When I dry-run the publish command, I get the following output:

```
pnpm -r publish --access public --no-git-checks --dry-run

npm notice 
npm notice 📦  @vue-youtube/core@0.0.6
npm notice === Tarball Contents === 
npm notice 782B   ./dist/index.cjs
npm notice 1.1kB  LICENSE         
npm notice 782B   dist/index.cjs  
npm notice 16.5kB dist/index.d.ts 
npm notice 846B   dist/index.js   
npm notice 1.3kB  package.json 
```

This is already odd as the `index.cjs` file is included twice (with slightly different paths, which shouldn't affect the final package contents). Where it gets even more weird is when I use `npm` instead of `pnpm`:

```
npm publish --workspaces --access public --dry-run
npm notice 
npm notice 📦  @vue-youtube/core@0.0.6
npm notice === Tarball Contents === 
npm notice 274B  README.md      
npm notice 3.4kB dist/index.cjs 
npm notice 4.7kB dist/index.d.ts
npm notice 4.8kB dist/index.js  
npm notice 1.4kB package.json 
```

So, locally, all required files are built but packaged slightly differently. This shouldn't affect the final published package. The now-published `manager.d.ts` file is not present here. On the other hand, the GitHub Action produces a slightly different package. It continues to include the old package contents.

---

I must investigate further until I publish a new fixed `0.0.6` version. Sorry in advance for the delay.

**Comment by suterma** [comment]
I noticed in the github repo, the package.json still says version 0.0.4. Did you forget to increment the number?

**Comment by Techassi** [comment]
Oh yeah, I did miss bumping that version. Thanks for pointing that out. I don't know if it causes the weird behaviour. It would be strange if it does. As I said, I will investigate further and come back to you.

**Comment by Techassi** [comment]
So the pnpm issue is still not solved, however, the publishing seems to work. You can now try version `0.0.6`. It seems to include everything you need. Let me know if everything works now :)

**Comment by suterma** [comment]
You have bumped the dependencies, as I see from package.json. While this is generally a good idea, I now get dependency issues when building my project.

```
D:\masu-data\github-suterma\replayer-pwa [main ≡]> npm i   
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: replayer@2.0.0
npm ERR! Found: @vue-youtube/core@0.0.4
npm ERR! node_modules/@vue-youtube/core
npm ERR!   @vue-youtube/core@"^0.0.6" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! @vue-youtube/core@"^0.0.6" from the root project  
npm ERR!
npm ERR! Conflicting peer dependency: vue@3.3.7
npm ERR! node_modules/vue
npm ERR!   peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.2
npm ERR!   node_modules/@vue/composition-api
npm ERR!     peerOptional @vue/composition-api@"^1.7.2" from @vue-youtube/core@0.0.6
npm ERR!     node_modules/@vue-youtube/core
npm ERR!       @vue-youtube/core@"^0.0.6" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```
This is because my project still uses ` "vue": "^3.2.47",` It should not be a problem for me, more of a gentle reminder to bump vue at my side too. However, some people might not be able or willing to change their other dependencies.

I suggest you reverse your vue dependency to some lower version, mabe even just 3.0.0?

**Comment by Techassi** [comment]
The `pnpm` issue is now resolved. I might to a `0.0.7` release with the fixed packaging.

Regarding your dependency issue: Yes I will take a look at it and maybe lower the minimum required version of Vue.

**Comment by Techassi** [comment]
I will close this issue as the original request was implemented. I opened a new issue about the minimum required Vue version.

**Comment by suterma** [comment]
This now works great. I use deferred loading with the `autoLoad: true,` setting. 

I do not even need explicit loading: as long as I do not render the component containing the actual player div and player imports, no connection to Google's properties is made. 

Thanks for implementing this.
